core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -122,11 +146,14 @@ public void updateEntityAttachments(long userIdentityId,
       }
     });
 
-    List<AttachmentContextEntity> existingAttachmentsContext =
-                                                             attachmentStorage.getAttachmentContextByEntity(entityId, entityType);
-    List<String> existingAttachmentsIds = existingAttachmentsContext.stream()
-                                                                    .map(AttachmentContextEntity::getAttachmentId)
-                                                                    .collect(Collectors.toList());
+    List<Attachment> existingEntityAttachments = null;
+    try {
+      existingEntityAttachments = attachmentStorage.getAttachmentsByEntity(userIdentityId, entityId, entityType);
+    } catch (NotFoundException | RepositoryException e) {
+      throw new ItemNotFoundException("Error while trying to get attachments of the entity with type" + entityType + "and id"
```

Comment:
Throwing a checked exception of JCR storage layer isn't convenient here IMO. You can define your own checked exceptions if needed